isis/src/base/apps/stretch/stretch.cpp

```diff
@@ -0,0 +1,88 @@
+#include "TextFile.h"
+#include "Statistics.h"
+#include "ProcessByLine.h"
+#include "SpecialPixel.h"
+#include "Stretch.h"
+#include "PvlGroup.h"
+#include "PvlKeyword.h"
+
+#include "stretch.h"
+
+namespace Isis {
+  void stretchProcess(Buffer &in, Buffer &out);
+  Stretch str;
+  Statistics stats;
+
+  void stretch(UserInterface &ui, Pvl *log) {
+    Cube *cubeFile = new Cube(ui.GetFileName("FROM"), "r");
```

Comment:
I'm fairly sure this will ignore cube attributes such as band selection. See how cubes are opened in cam2map:

https://github.com/USGS-Astrogeology/ISIS3/blob/dev/isis/src/base/apps/cam2map/cam2map.cpp#L24

also, you can read about what users can specify via cube attributes here:

https://isis.astrogeology.usgs.gov/documents/CommandLine/CommandLine.html#Cube%20Attributes